packages/default-theme/src/logic/useDomains.js

```diff
@@ -1,34 +1,69 @@
 import { computed } from "@vue/composition-api"
-import { getApplicationContext } from "@shopware-pwa/composables"
+import {
+  getApplicationContext,
+  useCms,
+  useNotifications,
+} from "@shopware-pwa/composables"
+import { getSeoUrls } from "@shopware-pwa/shopware-6-client"
+
+// the "last-chance route always has a name starting with `all_` - nuxt default"
+const PAGE_RESOLVER_ROUTE_PREFIX = "all_"
 
 export const useDomains = (rootContext) => {
-  const { router, routing } = getApplicationContext(rootContext, "useDomains")
+  const { router, routing, route, apiInstance } = getApplicationContext(
+    rootContext,
+    "useDomains"
+  )
 
   const availableDomains = computed(() => routing.availableDomains || [])
   const currentDomainId = computed(() => routing.getCurrentDomain().domainId)
+  const trimDomain = (url) => url.replace(routing.getCurrentDomain().url, "")
+  const getPathWithoutDomain = () => trimDomain(rootContext.$route.fullPath)
+
+  const isRouteStatic = () => !route.name.startsWith(PAGE_RESOLVER_ROUTE_PREFIX)
+  const getNewDomainUrl = async (domain) => {
+    let url = `${domain.url !== "/" ? `${domain.url}` : ""}`
+    let path = ""
+
+    if (isRouteStatic()) {
+      path += `/${getPathWithoutDomain()}`.replace(/\/\/+/, "/")
+    } else {
+      const { categoryId } = useCms(rootContext)
```

Comment:
Should apply to product detail pages and category pages equally.

Reply:
yes, I know that, categoryId points to resourceIndentifier, so it's working. I started to refactor the name of categoryId but it affects too many places... the PR would grow to 200 files, for real :)

Reply:
Okay, maybe a comment will do to have less confusion during future refactorings.

Reply:
I left the appropriate comment, and also create an issue with TODO to refactor the useCms composable because of the misleading naming :)